test/facilitator/approveBountyAmount.js

```diff
@@ -0,0 +1,234 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+const chai = require('chai');
+const sinon = require('sinon');
+const Web3 = require('web3');
+const Facilitator = require('../../libs/Facilitator/Facilitator');
+
+const assert = chai.assert;
+
+describe('Facilitator.approveBountyAmount()', () => {
+  let facilitator;
+  let web3;
+  let gatewayAddress;
+  let coGatewayAddress;
+  let baseTokenAddress;
+  let facilitatorAddress;
+  let bountyAmount;
+
+  let mockBaseTokenContract;
+  let mockTx;
+  let spy;
+
+  const setup = function() {
+    // Mock facilitator.getBounty method to return expected bounty amount.
+    sinon.stub(facilitator, 'getBounty').callsFake(() => {
+      return bountyAmount;
+    });
+
+    // Mock facilitator.getBaseToken method to return expected base token address.
+    sinon.stub(facilitator, 'getBaseToken').callsFake(() => {
+      return baseTokenAddress;
+    });
+
+    // Mock an instance of BaseToken contract.
+    mockBaseTokenContract = sinon.mock(
+      facilitator.contracts.BaseToken(baseTokenAddress)
+    );
+    const baseTokenContract = mockBaseTokenContract.object;
```

Comment:
I might be missing something here, but it looks like the `mockBaseTokenContract` is never used?